src/actions.ts

```diff
@@ -333,6 +333,29 @@ export function getActions(instance: PtzOpticsInstance): CompanionActionDefiniti
 				void instance.sendCommand(PresetRecall, event.options)
 			},
 		},
+		[PtzOpticsActionId.RecallPresetFromVar]: {
+			name: 'Recall Preset From Variable',
+			options: [
+				{
+					type: 'textinput',
+					label: 'Preset Number from Variable',
+					id: 'recallPresetVariableVal',
+					useVariables: true,
+					tooltip: 'Preset number range of 0-89, 100-254',
+					default: '0',
+				},
+			],
+			callback: async (event: CompanionActionEvent) => {
+				const varPreset = await instance.parseVariablesInString(event.options.recallPresetVariableVal as string)
+				const hexval = PresetRecallOption.valmap.get(parseInt(varPreset, 10))
+				if (hexval === undefined) {
+					instance.log('error', 'Invalid recall preset value of: ' + varPreset)
+					return
+				}
+				event.options.val = hexval
+				void instance.sendCommand(PresetRecall, event.options)
```

Comment:
You can just do

```typescript
    void instance.sendCommand(PresetRecall, { val: hexval })
```

rather than have to mutate/reuse the passed-in options.